tests/components/template/test_alarm_control_panel.py

```diff
@@ -0,0 +1,556 @@
+
+    hass.states.async_set("alarm_control_panel.test", STATE_ALARM_ARMED_AWAY)
+    await hass.async_block_till_done()
+
+    await common.async_alarm_arm_away(
+        hass, entity_id="alarm_control_panel.test_template_panel"
+    )
+    state = hass.states.get("alarm_control_panel.test_template_panel")
+    await hass.async_block_till_done()
+    assert state.state == STATE_ALARM_ARMED_AWAY
+
+    await common.async_alarm_arm_home(
+        hass, entity_id="alarm_control_panel.test_template_panel"
+    )
+    state = hass.states.get("alarm_control_panel.test_template_panel")
+    await hass.async_block_till_done()
+    assert state.state == STATE_ALARM_ARMED_AWAY
+
+    await common.async_alarm_arm_night(
+        hass, entity_id="alarm_control_panel.test_template_panel"
+    )
+    state = hass.states.get("alarm_control_panel.test_template_panel")
+    await hass.async_block_till_done()
+    assert state.state == STATE_ALARM_ARMED_AWAY
+
+    await common.async_alarm_disarm(
+        hass, entity_id="alarm_control_panel.test_template_panel"
+    )
+    state = hass.states.get("alarm_control_panel.test_template_panel")
+    await hass.async_block_till_done()
+    assert state.state == STATE_ALARM_ARMED_AWAY
+
+
+async def test_template_syntax_error(hass, caplog):
+    """Test templating syntax error."""
+    await setup.async_setup_component(
+        hass,
+        "alarm_control_panel",
+        {
+            "alarm_control_panel": {
+                "platform": "template",
+                "panels": {
+                    "test_template_panel": {
+                        "value_template": "{% if blah %}",
+                        "arm_away": {
+                            "service": "alarm_control_panel.alarm_arm_away",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "arm_home": {
+                            "service": "alarm_control_panel.alarm_arm_home",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "arm_night": {
+                            "service": "alarm_control_panel.alarm_arm_night",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "disarm": {
+                            "service": "alarm_control_panel.alarm_disarm",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                    }
+                },
+            }
+        },
+    )
+
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 0
+    assert ("invalid template") in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4788050)